src/auth_middleware.rs

```diff
@@ -199,11 +204,7 @@ impl AroundMiddleware for AuthMiddleware {
 
 impl AuthMiddleware {
     pub fn verify(token: &str, auth_db_file: &str) -> Result<(), ()> {
-        if token.is_empty() {
-            return Err(());
-        }
-
-        let token = match Token::<Header, SessionClaims>::parse(token) {
+        let token = match SessionToken::from_string(token) {
             Ok(token) => token,
             Err(_) => return Err(())
         };
```

Comment:
just a note that if you want to use try! you could add a From implementation. https://doc.rust-lang.org/book/error-handling.html has been a useful read to me :)